api/admin/controller/custom_lists.py

```diff
@@ -180,42 +180,41 @@ def _create_or_update_list(
             if auto_update_query is not None:
                 try:
                     auto_update_query_str = json.dumps(auto_update_query)
-                except json.JSONDecodeError:
-                    raise Exception(
+                except TypeError:
```

Comment:
This is updated from `JSONDecodeError` to `TypeError`, because previously it had no test coverage and when I did this update I decided to add a test here. It turns out that it was catching the wrong exception. `json.dumps` doesn't raise `JSONDecodeError`.